specs/protocol/holocene/exec-engine.md

```diff
@@ -73,5 +81,73 @@ an outbound withdrawal for a long period of time, the node may not have access t
 clients are able to at the very least reconstruct the account storage root at a given block on the fly if it does not
 directly store this information.
 
+## Extended `PayloadAttributesV3`
+
+The [`PayloadAttributesV3`](https://github.com/ethereum/execution-apis/blob/cea7eeb642052f4c2e03449dc48296def4aafc24/src/engine/cancun.md#payloadattributesv3)
+type is extended to:
+
+```rs
+PayloadAttributesV3: {
+    timestamp: QUANTITY
+    random: DATA (32 bytes)
+    suggestedFeeRecipient: DATA (20 bytes)
+    withdrawals: array of WithdrawalV1
+    parentBeaconBlockRoot: DATA (32 bytes)
+    transactions: array of DATA
+    noTxPool: bool
+    gasLimit: QUANTITY or null
+    eip1559Params: QUANTITY or null
```

Comment:
Instead of QUANTITY shouldn't this just be DATA (8 bytes)?

Reply:
I also don't really like the "or null" here, or for gaslimit above, where it actually should never be null. It is an optional parameter and can be omitted by the L1, but should always be specified by an L2.

I see you are using an explicit "null" value to indicate "use the chain config" but "null" vs "unspecified because it's optional" gets a little ambiguous and could probably use clarification.  Ideally we just wouldn't allow null if the parameter is specified, and we can mark it mandatory for all Holocene  blocks (including the first one). Perhaps just use the all-0 8-byte array as the initializer?

Reply:
I'm open to using the "all-0 8 byte array" as the initializer instead!